packages/unified-selection/src/unified-selection/iModel/IModel.ts

```diff
@@ -0,0 +1,136 @@
+ */
+export interface SelectRemoveEvent {
+  /** The type of operation that produced this event. */
+  type: SelectionSetEventType.Remove | SelectionSetEventType.Clear;
+  /** The element Ids removed from the set. */
+  removed: IdArg;
+  /** The affected SelectionSet. */
+  set: SelectionSet;
+}
+
+/**
+ * Passed to [[SelectionSet.onChanged]] event listeners when elements are simultaneously added to and removed from the selection set.
+ */
+export interface SelectReplaceEvent {
+  /** The type of operation that produced this event. */
+  type: SelectionSetEventType.Replace;
+  /** The element Ids added to the set. */
+  added: IdArg;
+  /** The element Ids removed from the set. */
+  removed: IdArg;
+  /** The affected SelectionSet. */
+  set: SelectionSet;
+}
+
+/**
+ * Payload sent to [[SelectionSet.onChanged]] event listeners to describe how the contents of the set have changed.
+ */
+export type SelectionSetEvent = SelectAddEvent | SelectRemoveEvent | SelectReplaceEvent;
+
+/**
+ * Manages a set of *listeners* for a particular event and notifies them when the event is raised.
+ */
+export interface Event<T> {
+  /** Registers a Listener to be executed whenever this event is raised. */
+  addListener(listener: T, scope?: any): () => void;
+  /** Un-register a previously registered listener. */
+  removeListener(listener: T, scope?: any): boolean;
+}
+
+/**
+ * A set of *currently selected* elements for an IModelConnection.
+ */
+export interface SelectionSet {
+  /** iModel of the set. */
+  iModel: IModelConnection;
+  /** Event called whenever elements are added or removed from this SelectionSet. */
+  onChanged: Event<(ev: SelectionSetEvent) => void>;
+  /** The IDs of the selected elements. */
+  get elements(): Set<string>;
+  /** Clear current selection set. */
+  emptyAll(): void;
+  /** Add one or more Ids to the current selection set. */
+  add(elem: IdArg): boolean;
+  /** Remove one or more Ids from the current selection set. */
+  remove(elem: IdArg): boolean;
+}
+
+/**
+ * A set if ID's optimized for performance-critical code which represents large sets of ID's as pairs of 32-bit integers.
+ */
+export interface Uint32Set {
+  /** Add any number of Ids to the set. */
+  addIds(ids: IdArg): void;
+  /** Remove any number of Ids from the set. */
+  deleteIds(ids: IdArg): void;
+}
+
+/**
+ * A set of *hilited* elements for an [[IModelConnection]], by element id.
+ */
+export interface IModelHiliteSet {
+  /** Control whether the hilited elements will be synchronized with the contents of the [[SelectionSet]].*/
+  set wantSyncWithSelectionSet(want: boolean);
+  /** The set of hilited elements. */
+  get elements(): Uint32Set;
+  /** The set of hilited subcategories. */
+  subcategories: Uint32Set;
+  /** The set of hilited models. */
+  models: Uint32Set;
+  /** Remove all elements from the hilited set. */
+  clear(): void;
+}
+
+/**
+ * Interface representing a connection to iModel.
+ */
+export interface IModelConnection {
```

Comment:
I think this is the only type in this file that needs to be exported.

Reply:
`SelectionSetEvent` was also needed. Changed to no longer export the other types.